### DISCUSSION TITLE
Ethereum Provider configuration not working across wallet implementations

### DISCUSSION THREAD

**Discussion by kyle-layerzero** [WalletConnect/walletconnect-monorepo#3831] [Developer Support]
We have a dApp and are using `@walletconnect/ethereum-provider` to connect via WalletConnect. Based on the discussion [here](https://safe-global.notion.site/safe-global/WalletConnect-v2-update-Issues-and-solutions-for-smart-wallets-3fc32fad6af4485fa5823eaebd486819), we tried using only `optionalChains`, but found that some mobile wallets will not work without the default chain. So we are stuck choosing between supporting multisig wallets and not some mobile ones, or vice versa, or duplicating the WalletConnect options with two different configurations.

Is there a recommended approach to support all wallet implementations via one WalletConnect configuration?

Examples:
```
const worksWithMultisig = {
    projectId: '123',
    optionalChains: nativeChainIds,
    showQrModal: true,
}

const worksWithAllMobileWallets = {
    projectId: '123',
    chains: [1], // Use Ethereum as the default chain, some wallets require this
    optionalChains: nativeChainIds,
    showQrModal: true,
}
```

**Comment by glitch-txs** [comment]
Which wallets?

**Reply by kyle-layerzero** [reply]
An example that does not work with the default chain is [SafePal](https://www.safepal.com/en/) on iOS. A multisig that does not work with the default chain is [Safe{Wallet}](https://safe.global/wallet).

**Reply by glitch-txs** [reply]
We are aware of this and we're currently communicating with wallets to fix their support to this configuration, we'll reach out SafePal again. This is why we haven't updated our Wagmi connector yet, but almost all major wallets are already supporting this configuration

**Reply by glitch-txs** [reply]
if you are able to upgrade to Web3Modal SDK you can use the Networks view which will let the user choose the required chain, for ethereum provider you can dynamically pass the chain to the connect function
